src/datasources/products/components/query-builder/utils/keyValueOperationUtils.ts

```diff
@@ -0,0 +1,136 @@
+import { FilterExpressions, FilterOperations } from "core/query-builder.constants";
+import { PropertyFieldKeyValuePair } from "datasources/products/types";
+
+
+export function editorTemplate(_: string, value: PropertyFieldKeyValuePair): HTMLElement {
+    const keyPlaceholder = `Key`;
+    const valuePlaceholder = `Value`;
+
+    const template = `
+        <div id="sl-query-builder-key-value-editor">
+            <ul style="list-style: none; padding-left: 0; padding-right: 10px;">
+                <li>
+                    <smart-input class="key-input" style="width: auto; padding-left: 5px;"
+                        placeholder="${keyPlaceholder}"
+                        value="${value?.key ?? ''}">
+                    </smart-input>
+                </li>
+                <li>
+                    <smart-input class="value-input" style="width: auto; margin-top: 10px; padding-left: 5px;"
+                        placeholder="${valuePlaceholder}"
+                        value="${value?.value ?? ''}">
+                    </smart-input>
+                </li>
+            </ul>
+        </div>`;
+
+    const templateBody = new DOMParser().parseFromString(template, 'text/html').body;
+    return templateBody.querySelector('#sl-query-builder-key-value-editor')!;
+}
+
+export function valueTemplate(editor: HTMLElement | null | undefined, value: PropertyFieldKeyValuePair): string {
+    if (value) {
+        const keyValuePair = value;
+        return `${keyValuePair.key} : ${keyValuePair.value}`;
+    }
+    if (editor) {
+        const keyInput = editor.querySelector<HTMLInputElement>('.key-input');
+        const valueInput = editor.querySelector<HTMLInputElement>('.value-input');
+        if (keyInput && valueInput) {
+            return `${keyInput.value} : ${valueInput.value}`;
+        }
+    }
+    return '';
+}
+
+export function handleStringValue(editor: HTMLElement | null | undefined): { label: PropertyFieldKeyValuePair; value: PropertyFieldKeyValuePair; } {
+    const inputs = retrieveKeyValueInputs(editor);
+    return {
+        label: inputs,
+        value: inputs
+    };   
```

Comment:
```suggestion
    };
```
nit: Seems like there are unnecessary spaces at the end of lines across different files - can you look and fix them all?

Reply:
removed the space